plugins/colorschemes/modus.nix

```diff
@@ -0,0 +1,93 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "modus";
+  isColorscheme = true;
+  originalName = "modus-themes-nvim";
+  defaultPackage = pkgs.vimPlugins.modus-themes-nvim;
+
+  maintainers = [ lib.maintainers.nwjsmith ];
+
+  settingsOptions = {
+    style =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "modus_operandi"
+          "modus_vivendi"
+        ]
+        ''
+          The theme comes in two styles, `modus_operandi`, a dark variant `modus_vivendi`.
+        '';
+
+    variant =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "default"
+          "tinted"
+          "deuteranopia"
+          "tritanopia"
+        ]
+        ''
+          The theme comes in four variants `default`, `tinted`, `deuteranopia`, and `tritanopia`.
+        '';
+
+    transparent = helpers.defaultNullOpts.mkBool false ''
+      Disable setting the background color.
+    '';
+
+    dim_inactive = helpers.defaultNullOpts.mkBool false ''
+      Dims inactive windows.
+    '';
+
+    styles = {
+      comments = helpers.defaultNullOpts.mkHighlight "{italic = true;}" "" ''
+        Define comments highlight properties.
+      '';
+
+      keywords = helpers.defaultNullOpts.mkHighlight "{italic = true;}" "" ''
+        Define keywords highlight properties.
+      '';
+
+      functions = helpers.defaultNullOpts.mkHighlight "{}" "" ''
+        Define functions highlight properties.
+      '';
+
+      variables = helpers.defaultNullOpts.mkHighlight "{}" "" ''
+        Define variables highlight properties.
+      '';
+    };
+
+    on_colors = helpers.defaultNullOpts.mkLuaFn "function(colors) end" ''
+      Override specific color groups to use other groups or a hex color.
+      Function will be called with a `ColorScheme` table.
+      `@param colors ColorScheme`
+    '';
+
+    on_highlights = helpers.defaultNullOpts.mkLuaFn "function(highlights, colors) end" ''
+      Override specific highlights to use other groups or a hex color.
+      Function will be called with a `Highlights` and `ColorScheme` table.
+      `@param highlights Highlights`
+      `@param colors ColorScheme`
+    '';
+  };
+
+  settingsExample = {
+    style = "auto";
+    variant = "default";
+    transparent = false;
+    dim_inactive = false;
+    styles = {
+      comments.italic = true;
+      keywords.italic = true;
+      functions = { };
+      variables = { };
+    };
+    on_colors = "function(colors) end";
+    on_highlights = "function(highlights, colors) end";
+  };
+}
```

Comment:
Most colorschemes set `opts.termguicolors = mkDefault true` in `extraConfig`. Does this plugin need `termguicolors`?

Reply:
Good catch, it is required. It's based on tokyonight which also requires it. Fixed.